### DISCUSSION TITLE
How to define command-level option when using multicommand class?

### DISCUSSION THREAD

**Discussion by pySilver** [django-commons/django-typer#134] [Q&A] [answered]
First of all: Thanks for such a great django wrapper for typer!

I want to add an option callback to my command class. The problem is that I have multiple commands in that class:

```
class Command(TyperCommand):
    @command()
    def foo(): ...
    
    @command
    def bar(): ...
```

and my option belongs to whole command (just like `--version` or `--settings`) rather than specific command. How do I do it?

**Comment by bckohan** [comment, marked as answer]
Hi @pySilver - Thanks! I moved this over to discussions because I'm sure others will have the same question.

You can use either [@callback or @initialize](https://django-typer.readthedocs.io/en/latest/howto.html#define-an-initialization-callback) (initialize is just an alias for callback that I thought made more sense for class based commands):

```python
from django_typer.management import TyperCommand, initialize, command

class Command(TyperCommand):

    @initialize()
    def init(): ...

    @command()
    def foo(): ...
    
    @command()
    def bar(): ...
```

You can add any common options shared by foo and bar to ``init``. ``init`` will be invoked first with those options. If you add options that share names with the Django default options you will override them. If you want to suppress any Django default options you can [use suppressed_base_arguments](https://django-typer.readthedocs.io/en/latest/howto.html#change-default-django-options) as with vanilla BaseCommand classes.

``@initialize`` is a wrapper for ``@app.callback()`` so it takes all the same arguments, for instance if you want to be able to invoke ``init`` without ``foo`` or ``bar`` [pass ``invoke_without_command=True``](https://click.palletsprojects.com/en/latest/commands/#group-invocation-without-command).

**Reply by pySilver** [reply]
Oh! Great. Thanks for your help!
